nexus/db-queries/src/db/datastore/vpc.rs

```diff
@@ -861,9 +854,23 @@ impl DataStore {
         assert_eq!(authz_vpc.id(), subnet.vpc_id);
 
         let db_subnet = self.vpc_create_subnet_raw(subnet).await?;
-        self.vpc_system_router_ensure_subnet_routes(opctx, authz_vpc.id())
+
+        let route_id = route_id.unwrap_or_else(Uuid::new_v4);
+        let route = db::model::RouterRoute::new_subnet(
+            route_id,
+            authz_system_router.id(),
+            db_subnet.name().clone().into(),
+            db_subnet.id(),
+        );
+
+        self.router_create_route(opctx, authz_system_router, route)
```

Comment:
Just to confirm:

The spots where we are replacing `vpc_system_router_ensure_subnet_routes` with more fine-grained updates to the database appear to be all done outside transactions. This means the updates will appear non-atomically.

In this case, I think that means "route creation + incrementing the RPW version" are not updated in lock-step. Is that okay?

Reply:
Thanks for the quick review, Sean.

They don't need to happen in lockstep (the RPW version bump has always fallen outside of the transaction), but they do need to be ordered in a fairly strict way. State updates need to bump the version *after* they have been applied. The routing RPW reads a version for each router (say, version X), and then computes a reified set of rules that are guaranteed to contain state of _at least version X_. Any missing or broken links are treated as no-op rules.